### PR TITLE
Lime 76, Issue 38, Named return issues resolved

### PR DESCRIPTION
**Pull request by saxenism**
## Description
If a function returns something and in the function signature, the name of the returning variable along with the type is mentioned, it is a good practice to use that name while returning a value from the function. 
 
PR opened to fix issue at: https://github.com/code-423n4/2021-12-sublime-findings/issues/38

## Change Log
With this edit of named return we implemented an improvement in code styling/clarity.

### CODE REVIEW

**Comment by prateekreddy** [comment]
Has this style been consistent across all functions ?

**Review by prateekreddy — CHANGES_REQUESTED** [review]
@saxenism Please check if this style been consistent across all functions ? Let's ensure we use return everywhere even in case of multiple return value

**Comment by saxenism** [comment]
> @saxenism Please check if this style been consistent across all functions ? Let's ensure we use return everywhere even in case of multiple return value

Understood. Switched the return style. Please review again.

**Review by ritik99 — COMMENTED** [review]
@saxenism I don't think the styling is still consistent throughout the contracts. Some instances: [1](https://github.com/sublime-finance/sublime-v1/blob/f25860e182e4c9a8fed0332f1cbd4755d68b3af4/contracts/Pool/Pool.sol#L244), [2](https://github.com/sublime-finance/sublime-v1/blob/f25860e182e4c9a8fed0332f1cbd4755d68b3af4/contracts/Pool/Pool.sol#L165). Please recheck all the contracts

**Comment by saxenism** [comment]
> @saxenism I don't think the styling is still consistent throughout the contracts. Some instances: [1](https://github.com/sublime-finance/sublime-v1/blob/f25860e182e4c9a8fed0332f1cbd4755d68b3af4/contracts/Pool/Pool.sol#L244), [2](https://github.com/sublime-finance/sublime-v1/blob/f25860e182e4c9a8fed0332f1cbd4755d68b3af4/contracts/Pool/Pool.sol#L165). Please recheck all the contracts

Scanned all contracts and made the return style consistent. Please check again.

Few functions that I did not change:
1. [SavingsAccount.withdrawAll](https://github.com/sublime-finance/sublime-v1/blob/master/contracts/SavingsAccount/SavingsAccount.sol#L304)

This is because I did not see the function returning at all. Maybe we can discuss the function's return declaration. 

2. [PoolFactory._deploy](https://github.com/sublime-finance/sublime-v1/blob/master/contracts/Pool/PoolFactory.sol#L404)

This is because I didn't really want to change a function making an assembly call.

3.[ NoYield.emergencyWithdraw](https://github.com/sublime-finance/sublime-v1/blob/master/contracts/yield/NoYield.sol#L78)

This is because the function's logic is to be changed from another PR, and style changes can be made after that.

**Comment by prateekreddy** [comment]
@saxenism IsavingsAccount has some inconsistency with the format. Can you check ?

**Comment by prateekreddy** [comment]
In all the interfaces let's name the return values, so it is more clear. We need not do this on actual functions though.

**Comment by saxenism** [comment]
> @saxenism In all the interfaces let's name the return values, so it is more clear. We need not do this on actual functions though.
> 
> PoolFactory, SavingsAccount, NoYield has inconsistent return patterns. Please check.

Made the required changes, apart from `NoYield.emergencyWithdraw` because that function has yet to be changed.
Please review again.

File: contracts/SavingsAccount/SavingsAccountUtil.sol [hunk 1]
```diff
@@ -47,11 +47,10 @@ library SavingsAccountUtil {
         address _strategy,
         address _from,
         address _to,
-        uint256 _amount,
-        address _token,
-        address _strategy
+        uint256 _amount
     ) internal returns (uint256) {
-        transferTokens(_token, _amount, _from, address(this));
+        uint256 _sharesReceived;
```
Comment (prateekreddy) [inline contracts/SavingsAccount/SavingsAccountUtil.sol:52]: This can be declared at initialization ?

File: contracts/interfaces/IPool.sol [hunk 2]
```diff
@@ -94,13 +94,13 @@ interface IPool {
 
     function withdrawBorrowedAmount() external;
 
-    function borrower() external returns (address);
+    function borrower() external returns (address poolBorrower);
 
-    function getMarginCallEndTime(address _lender) external returns (uint256);
+    function getMarginCallEndTime(address _lender) external returns (uint256 marginCallEndTimeFor_lender);
```
Comment (prateekreddy) [inline contracts/interfaces/IPool.sol:99]: Inconsistent naming convention

File: contracts/interfaces/IPriceOracle.sol [hunk 3]
```diff
@@ -24,7 +24,7 @@ interface IPriceOracle {
      */
     event UniswapPriceAveragingPeriodUpdated(uint32 uniswapPriceAveragingPeriod);
 
-    function getLatestPrice(address num, address den) external view returns (uint256, uint256);
+    function getLatestPrice(address num, address den) external view returns (uint256 priceOfNumInTermsOfDen, uint256 priceDecimals);
```
Comment (prateekreddy) [inline contracts/interfaces/IPriceOracle.sol:27]: (uint256 price, uint256 decimals) should be sufficient ?

File: contracts/interfaces/IStrategyRegistry.sol [hunk 4]
```diff
@@ -20,9 +20,9 @@ interface IStrategyRegistry {
      */
     event MaxStrategiesUpdated(uint256 maxStrategies);
 
-    function registry(address _strategy) external view returns (bool);
+    function registry(address _strategy) external view returns (bool isValidStrategy);
 
-    function getStrategies() external view returns (address[] memory);
+    function getStrategies() external view returns (address[] memory strategiesArray);
```
Comment (prateekreddy) [inline contracts/interfaces/IStrategyRegistry.sol:25]: strategies should be sufficient

File: contracts/yield/NoYield.sol [hunk 5]
```diff
@@ -48,7 +48,7 @@ contract NoYield is IYield, Initializable, OwnableUpgradeable, ReentrancyGuard {
     /**
      * @notice used to query liquidity token for a given asset
      * @param _asset address of the asset
-     * @return _tokenAddress: address of the lqiudity token for the asset
+     * @return _asset: address of the lqiudity token for the asset
```
Comment (prateekreddy) [inline contracts/yield/NoYield.sol:51]: return value and param have same var name

**Review by prateekreddy — CHANGES_REQUESTED** [review]

File: contracts/yield/NoYield.sol [hunk 6]
```diff
@@ -48,10 +48,11 @@ contract NoYield is IYield, Initializable, OwnableUpgradeable, ReentrancyGuard {
     /**
      * @notice used to query liquidity token for a given asset
      * @param _asset address of the asset
-     * @return _asset: address of the lqiudity token for the asset
+     * @return _tokenAddress: address of the lqiudity token for the asset
      **/
     function liquidityToken(address _asset) external pure override returns (address) {
-        return _asset;
+        address _tokenAddress = _asset;
```
Comment (prateekreddy) [inline contracts/yield/NoYield.sol:54]: you can return asset directly, assigning to tokenAddress is not necessary

**Review by prateekreddy — CHANGES_REQUESTED** [review]

**Review by prateekreddy — APPROVED** [review]

**Pull request merged**